test/Database/LSMTree/Class/Normal.hs

```diff
@@ -169,55 +213,126 @@ withTableOpen ::
 withTableOpen sesh snap = bracket (open sesh snap) close
 
 withTableDuplicate ::
-     forall h m k v blob a. (IOLike m, IsTableHandle h)
+     forall h m k v blob a. ( IOLike m, IsTableHandle h
+      , M2.C k v blob
+      )
   => h m k v blob
   -> (h m k v blob -> m a)
   -> m a
 withTableDuplicate table = bracket (duplicate table) close
 
 withCursor ::
-     forall h m k v blob a. (IOLike m, IsTableHandle h, SerialiseKey k)
+     forall h m k v blob a. (IOLike m, IsTableHandle h, SerialiseKey k
+     , M2.C k v blob
+     )
   => Maybe k
   -> h m k v blob
   -> (Cursor h m k v blob -> m a)
   -> m a
 withCursor offset hdl = bracket (newCursor offset hdl) (closeCursor (Proxy @h))
 
 {-------------------------------------------------------------------------------
-  Model instance
+  Model 2 instance
```

Comment:
I added a short module description for each model, but it could probably still need some more documentation.

> What I want to see somewhere in these models and instances is a high level explanation of them all.

Not sure where best to put that, but it could be make use of the comparison in https://github.com/IntersectMBO/lsm-tree/pull/403#issuecomment-2426564132.